spaceballs_ws_server/can_server.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python3
+from sys import exit
+from signal import signal, SIGINT
+
+import asyncio
+import can
+from spaceballs_ws_server.ws_data_server import WebSocketDataServer
+
+
+class CANServer:
+
+    def quit(self, signal_received, frame):
+        # Handle any cleanup here
+        print('SIGINT or CTRL-C detected. Exiting gracefully')
+        self.loop.stop()
+        self.bus.shutdown()
+        self.bus2.shutdown()
+
+        for t in self.tasks:
+            t.stop()
+
+        exit(0)
+
+    def __init__(self, config_path):
+        #virtual interfaces for development
+        self.bus = can.interface.Bus('test', bustype='virtual')
+
+        self.bus2 = can.interface.Bus('test', bustype='virtual')
+
+        self.reader = can.AsyncBufferedReader()
+
+        self.listeners = [
+            self.reader
+        ]
+
+        self.tasks = [] # stores messges being sent
+
+        self.loop = asyncio.get_event_loop()
+
+        self.notifier = can.Notifier(self.bus, self.listeners, loop=self.loop)
+
+        self.web_socket_server = WebSocketDataServer(config_path, self.reader)
+
+    def run(self):
+        """
+        Run the server
+        """
+        self.loop.run_until_complete(self.web_socket_server.run(self.reader))
+        self.loop.run_forever()
+
+    def mockCanBus(self):
```

Comment:
Should maybe make the use of the virtual interface and mocked bus a parameter when development is further along and maybe load the messages from a config file from the tests folder.

Reply:
Deffo. I'll make this into a mock message in the test suite as a next step. I just ran out of time and I wanted to get the PR sent to you

Reply:
I've made an issue to add unit tests.